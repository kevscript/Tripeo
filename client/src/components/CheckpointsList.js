import React from 'react'
import styled from 'styled-components'
import CheckpointCard from './CheckpointCard'
import PropTypes from 'prop-types'

const ListContainer = styled.ul`
  width: 100%;
  list-style: none;
`

const CheckpointsList = ({ checkpoints }) => {
  return (
    <ListContainer data-test="List">
      {checkpoints.map((checkpoint) => {
        return (
          <CheckpointCard
            key={checkpoint.startDate}
            checkpoint={checkpoint}
          />
        )
      })}
    </ListContainer>
  )
}

CheckpointsList.propTypes = {
  checkpoints: PropTypes.array
}

export default CheckpointsList