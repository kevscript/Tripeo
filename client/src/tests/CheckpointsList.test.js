import React from 'react'
import CheckpointsList from '../components/CheckpointsList'
import { shallow } from 'enzyme'

describe('CheckpointsList component', () => {

  describe('when the list has items', () => {
    let props = {
      checkpoints: [
        { startDate: '01/02/2018' },
        { startDate: '04/02/2018' },
        { startDate: '07/02/2018' }],
    }
    let component
    let wrapper

    beforeEach(() => {
      component = shallow(<CheckpointsList {...props} />)
      wrapper = component.find('[data-test="List"]')
    })

    test('should contain a list', () => {
      expect(wrapper.length).toBe(1)
    })

    test('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    test('should render an item in the list for every checkpoint', () => {
      expect(wrapper.children()).toHaveLength(3)
    })

    test('should chronogically pass the right checkpoint to the right child item', () => {
      const children = wrapper.children()
      children.map((el, i) => {
        expect(el.prop('checkpoint')).toEqual(props.checkpoints[i])
      })
    })
  })

  describe('when the list is empty', () => {
    let props = {
      checkpoints: []
    }
    let component
    let wrapper

    beforeEach(() => {
      component = shallow(<CheckpointsList {...props} />)
      wrapper = component.find('[data-test="List"]')
    })

    test('should contain a list', () => {
      expect(wrapper.length).toBe(1)
    })

    test('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    test('the list shouldnt render without child element', () => {
      expect(wrapper.children()).toHaveLength(0)
    })
  })
})