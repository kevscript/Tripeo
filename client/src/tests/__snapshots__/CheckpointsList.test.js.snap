// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckpointsList component when the list has items should render correctly 1`] = `ShallowWrapper {}`;

exports[`CheckpointsList component when the list is empty should render correctly 1`] = `ShallowWrapper {}`;
